commons/src/test/scala/dagr/commons/io/IoTest.scala

```diff
@@ -132,4 +149,8 @@ class IoTest extends UnitSpec {
     val grandchild = PathUtil.pathTo(dir.toAbsolutePath.toString, "grand/child")
     Io.findFirstExtentParent(grandchild).get.toAbsolutePath.toString shouldBe dir.toAbsolutePath.toString
   }
+
+  it should "return None if no parent was found" in {
+    Io.findFirstExtentParent(PathUtil.pathTo("/")) shouldBe 'empty
```

Comment:
My point here, now that I think about it more, is that this test is testing under-specified behaviour that is a consequence of how the function is implemented.  Tests should test specified behaviour, and if things are under-specified we should fix that first, vs. creating tests that assert that the current, under-specified behaviour, is maintained.

In this case the method is effectively private - it's only `private[util]` so that it can be tested, and is never called with a path that exists.